src/validationinterface.h

```diff
@@ -173,6 +173,13 @@ class CValidationInterface {
      * Notifies listeners that a block which builds directly on our current tip
      * has been received and connected to the headers tree, though not validated yet */
     virtual void NewPoWValidBlock(const CBlockIndex *pindex, const std::shared_ptr<const CBlock>& block) {};
+    /**
+     * Notifies the validation interface that it is being unregistered
+     */
+    virtual void ValidationInterfaceUnregistering() {};
```

Comment:
You don't need it in https://github.com/bitcoin-core/gui/pull/444 but is it worth to consider to add `::ValidationInterfaceRegistering` as part of this PR for completeness?